jOOQ/src/main/java/org/jooq/impl/AbstractField.java

```diff
@@ -947,7 +947,7 @@ public final Condition in(Field<?>... values) {
 
     @Override
     public final Condition in(Collection<?> values) {
-        List<Field<?>> fields = new ArrayList<Field<?>>();
+        List<Field<?>> fields = new ArrayList<Field<?>>(values.size());
```

Comment:
This can be taken one step further, because the list is going to be converted to an array again, later. So why not just use an array, instead?